unity/Assets/Scripts/Quest/DialogWindow.cs

```diff
@@ -95,15 +95,32 @@ public void CreateWindow()
         }
 
         int num = 1;
+        var varManager = Game.Get().quest.vars;
         foreach (EventButton eb in buttons)
         {
+            // Handle condition failure
+            var buttonConditionFailed = eb.condition != null && !varManager.Test(eb.condition);
+            if (buttonConditionFailed && eb.conditionFailedAction == ButtonAction.HIDE)
+            {
+                continue;
+            }
             int numTmp = num++;
             ui = new UIElement();
             ui.SetLocation(hOffset, offset, buttonWidth, 2);
-            ui.SetText(eb.GetLabel(), eb.colour);
-            ui.SetFontSize(UIScaler.GetMediumFont());
-            ui.SetButton(delegate { onButton(numTmp); });
-            new UIElementBorder(ui, eb.colour);
+            if (buttonConditionFailed && eb.conditionFailedAction == ButtonAction.DISABLE)
+            {
+                ui.SetText(eb.GetLabel(), Color.gray);
+                ui.SetFontSize(UIScaler.GetMediumFont());
+                new UIElementBorder(ui, Color.gray);
+            }
```

Comment:
Should we expose a disabled button UI colors somewhere?

Reply:
Would be nice but this will also make it necessary to create even more UI controls in the editor. Therefore I would postpone it until someone needs this feature.